getting_started/scripting/gdscript/gdscript_types_intro.rst

```diff
@@ -0,0 +1,413 @@
+.. _doc_gdscript_types_intro:
+
+Get started with static types in GDscript
```

Comment:
With regards to our table of contents, I prefer short and to the point titles like "Static-typing in GDScript" than whole sentences. The TOC has a limited width, so it's better to use noun groups whenever it's explicit enough. See e.g. http://docs.godotengine.org/en/latest/tutorials/3d/index.html

I leave it to you whether the "intro" part is necessary (i.e., will there be other pages on static typing that would warrant this one to be just an "intro"?). Ideally I prefer if the rst filename also matches the page title, so something like `static_typing.rst` (maybe no need to repeat `gdscript` there which is in the folder name) maybe.